commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java

```diff
@@ -29,13 +32,14 @@
         "    }")
 @ResourceInfo(pluralName = "customer groups", pathElement = "customer-groups")
 @HasByIdGetEndpoint(javadocSummary = "Fetches a customer group based on a known ID.", includeExamples = "io.sphere.sdk.customergroups.queries.CustomerGroupByIdGetIntegrationTest#execution()")
+@HasByKeyGetEndpoint(javadocSummary = "Fetches a customer group based on a known key.", includeExamples = "io.sphere.sdk.customergroups.queries.CustomerGroupByKeyGetIntegrationTest#execution()")
 @HasCreateCommand(javadocSummary = "Creates a new customer group", includeExamples = "io.sphere.sdk.customergroups.commands.CustomerGroupCreateCommandIntegrationTest#execution()", interfaceContents = "static CustomerGroupCreateCommand of(final String groupName) {\n" +
         "        return of(CustomerGroupDraft.of(groupName));\n" +
         "    }")
-@HasUpdateCommand(javadocSummary = "Updates a customer group.")
-@HasDeleteCommand(javadocSummary = "Deletes a customer group.", includeExamples = "io.sphere.sdk.customergroups.commands.CustomerGroupDeleteCommandIntegrationTest#execution()")
+@HasUpdateCommand(javadocSummary = "Updates a customer group.",updateWithKey = true)
+@HasDeleteCommand(javadocSummary = "Deletes a customer group.", includeExamples = "io.sphere.sdk.customergroups.commands.CustomerGroupDeleteCommandByIdIntegrationTest#execution()", deleteWithKey = true)
 @HasQueryModel
-public interface CustomerGroup extends Resource<CustomerGroup> {
+public interface CustomerGroup extends Resource<CustomerGroup>, WithKey {
```

Comment:
Good catch! I always forget to add the `With...` interfaces, but they're quite useful.